tests/api-controller-acceptance-tests/apicontroller/component_test.go

```diff
@@ -0,0 +1,397 @@
+
+			updatedApi, err = ctx.awaitApiCreated(kymaClient, updatedApi, false, true)
+			So(err, ShouldBeNil)
+			So(updatedApi.ResourceVersion, ShouldNotBeEmpty)
+			So(updatedApi.Spec, ctx.ShouldDeepEqual, api.Spec)
+			So(updatedApi.Status.AuthenticationStatus.Resource.Uid, ShouldBeEmpty)
+
+			_, err = istioAuthClient.AuthenticationV1alpha1().Policies(namespace).Get(createdApi.Status.AuthenticationStatus.Resource.Name, metav1.GetOptions{})
+			So(err, ShouldNotBeNil)
+		})
+
+		Convey("create API with custom jwt configuration", func() {
+			t.Log("create API with custom jwt configuration")
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName, apiSecurityDisabled, true)
+			ctx.setCustomJwtAuthenticationConfig(api)
+
+			lastApi, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			defer ctx.cleanUpApi(kymaClient, lastApi, t, false)
+			So(err, ShouldBeNil)
+			So(lastApi, ShouldNotBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+
+			lastApi, err = ctx.awaitApiCreated(kymaClient, lastApi, true, true)
+			So(err, ShouldBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+			So(lastApi.Spec, ctx.ShouldDeepEqual, api.Spec)
+
+			policy, err := istioAuthClient.AuthenticationV1alpha1().Policies(namespace).Get(lastApi.Status.AuthenticationStatus.Resource.Name, metav1.GetOptions{})
+			expectedPolicy := ctx.policyFor(testId, api.Spec.Authentication[0].Jwt.Issuer)
+			So(err, ShouldBeNil)
+			So(policy.Spec, ctx.ShouldDeepEqual, expectedPolicy)
+		})
+
+		Convey("delete API and all its related resources", func() {
+			t.Log("delete API and all its related resources")
+
+			testId := ctx.generateTestId(testIdLength)
+			t.Logf("Running test: %s", testId)
+			api := ctx.apiFor(testId, domainName, apiSecurityDisabled, true)
+			authEnabled := true
+			api.Spec.AuthenticationEnabled = &authEnabled
+
+			lastApi, err := kymaClient.GatewayV1alpha2().Apis(namespace).Create(api)
+			defer ctx.cleanUpApi(kymaClient, lastApi, t, true)
+			So(err, ShouldBeNil)
+			So(lastApi, ShouldNotBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+
+			lastApi, err = ctx.awaitApiCreated(kymaClient, lastApi, true, true)
+			So(err, ShouldBeNil)
+			So(lastApi.ResourceVersion, ShouldNotBeEmpty)
+			So(lastApi.Spec, ctx.ShouldDeepEqual, api.Spec)
+			policy, err := istioAuthClient.AuthenticationV1alpha1().Policies(namespace).Get(lastApi.Status.AuthenticationStatus.Resource.Name, metav1.GetOptions{})
+			So(err, ShouldBeNil)
+			vs, err := istioNetClient.NetworkingV1alpha3().VirtualServices(namespace).Get(lastApi.Status.VirtualServiceStatus.Resource.Name, metav1.GetOptions{})
+			So(err, ShouldBeNil)
+
+			err = kymaClient.GatewayV1alpha2().Apis(namespace).Delete(lastApi.Name, &metav1.DeleteOptions{})
+			So(err, ShouldBeNil)
+
+			time.Sleep(5 * time.Second)
+
+			_, err = kymaClient.GatewayV1alpha2().Apis(namespace).Get(lastApi.Name, metav1.GetOptions{})
```

Comment:
Maybe we should create the function awaitApiDeleted for a consistency with create/update. What do you think?

Reply:
I think it's better how it is right now, because those three checks are strict requirements: api-controller must clean all dependent resources. Extracting them to different method would hide it and make it look like a cleanup rather then expectation.

Reply:
The same can be told about using awaitApiChanged, where you "hide" expectations after creating/updating API. IMO hiding that doesn't make the code unclear. But it is more a personal preference so let's leave it unchanged.